modules/apps/repository/repository-test/src/testIntegration/java/com/liferay/repository/test/RepositoryFactoryTest.java

```diff
@@ -31,29 +55,80 @@ public class RepositoryFactoryTest {
 	@ClassRule
 	@Rule
 	public static final AggregateTestRule aggregateTestRule =
-		new LiferayIntegrationTestRule();
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
 
 	@Before
 	public void setUp() throws Exception {
 		_group = GroupTestUtil.addGroup();
 	}
 
+	@Test
+	public void testAddFolderInRepositoryWhenSameRepositoryIdExistsInDifferentPartition()
+		throws Exception {
+
+		if (!DBPartition.isPartitionEnabled()) {
+			return;
+		}
+
+		Company company = CompanyTestUtil.addCompany();
+		com.liferay.portal.kernel.model.Repository repository1;
+
+		long companyId = CompanyThreadLocal.getCompanyId();
+		String userId = PrincipalThreadLocal.getName();
+
+		CompanyThreadLocal.setCompanyId(company.getCompanyId());
+
+		try {
+			User user = UserTestUtil.addUser(company);
+
+			PrincipalThreadLocal.setName(user.getUserId());
+
+			Group group = GroupTestUtil.addGroupToCompany(
+				company.getCompanyId());
+
+			repository1 = _addRepository(group, user.getUserId());
+
+			_repositoryFactory.createRepository(repository1.getRepositoryId());
+		}
+		finally {
+			CompanyThreadLocal.setCompanyId(companyId);
+			PrincipalThreadLocal.setName(userId);
+		}
+
+		com.liferay.portal.kernel.model.Repository repository2 = _addRepository(
+			_group, TestPropsValues.getUserId(), repository1.getRepositoryId());
+
+		Assert.assertEquals(
+			repository1.getRepositoryId(), repository2.getRepositoryId());
+
+		Repository repository = _repositoryFactory.createRepository(
+			repository2.getRepositoryId());
+
+		repository.addFolder(
```

Comment:
Aren't you missing an assertion here? You're not testing that the result is correct (i.e. folder.getCompanyId() == repository2.getCompanId() && folder.getCompanyId() != repository1.getCompanyId()).